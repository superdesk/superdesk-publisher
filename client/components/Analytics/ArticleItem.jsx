import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ArticleItem = ({ item, style }) => {
  return (
    <div className="sd-flex-table__row" style={style}>
      <div className="sd-flex-table__cell sd-flex-grow">
        <div className="sd-text__strong">
          <strong>{item.title}.</strong>
        </div>
        <div>
          <span className="sd-text__date-time sd-text__italic">Category: </span>
          <span className="sd-text__normal sd-margin-r--1">
            {item.route && item.route.name}
          </span>
          {item.authors && item.authors.length ? (
            <React.Fragment>
              <span className="sd-text__date-time sd-text__italic">
                Authors:{" "}
              </span>
              <span className="sd-text__normal sd-margin-r--1">
                {item.authors.map((author, index) =>
                  index > 0 ? (
                    <span key={"author_" + index}>{author.name}, </span>
                  ) : (
                    <span key={"author_" + index}>{author.name} </span>
                  )
                )}
              </span>
            </React.Fragment>
          ) : null}

          <span className="sd-text__date-time sd-text__italic ">
            Published:{" "}
          </span>
          <time>{moment(item.published_at).format("YYYY.MM.DD, HH:mm")}</time>
        </div>
      </div>
      <div className="sd-flex-table__cell sd-text__center">
        {item._links && item._links.online && item._links.online.href ? (
          <a
            className="icn-btn"
            target="_blank"
            href={
              item.tenant.subdomain
                ? "http://" +
                  item.tenant.subdomain +
                  "." +
                  item.tenant.domain_name +
                  item._links.online.href
                : "http://" + item.tenant.domain_name + item._links.online.href
            }
          >
            <i className="icon-external" />
          </a>
        ) : null}
      </div>
      <div className="sd-flex-table__cell">
        {item.article_statistics.page_views_number}
      </div>
      {/* <div className="sd-flex-table__cell">
        {item.article_statistics.internal_click_rate}
      </div>
      <div className="sd-flex-table__cell">
        {item.article_statistics.impressions_number}
      </div> */}
    </div>
  );
};

ArticleItem.propTypes = {
  item: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired
};

export default ArticleItem;
