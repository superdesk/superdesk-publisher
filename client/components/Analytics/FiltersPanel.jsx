import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import moment from "moment";
import { Button, IconButton } from "superdesk-ui-framework/react";
import MultiSelect from "../UI/MultiSelect";
import AsyncMultiSelect from "../UI/AsyncMultiSelect";
import { DatePicker } from "superdesk-ui-framework/react";
import {gettext} from '../../superdeskApi';

class FiltersPanel extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      filters: props.filters ? props.filters : { route: [], author: [] },
      authors: [],
      dateFilterType: "range",
    };
  }

  componentDidUpdate(prevProps) {
    if (!_.isEqual(this.props.filters, prevProps.filters)) {
      let filters = { ...this.props.filters };
      let newRoutes = [];

      if (filters.routes && filters.routes.length) {
        filters.routes.forEach((item) => {
          let route = this.props.routes.find((a) => a.id === item);

          if (route) {
            newRoutes.push({
              value: route.id,
              label: route.name,
            });
          }
        });

        filters.routes = newRoutes;
      }

      this.setState({ filters: filters });
    }
  }

  loadAuthors = (inputValue = null) => {
    if (inputValue && inputValue.length < 3) return this.state.authors;

    return this.props.publisher
      .queryAuthors({ term: inputValue, limit: 30 })
      .then((response) => {
        let authorsOptions = [];

        response._embedded._items.forEach((item) => {
          authorsOptions.push({
            value: item.id,
            label: item.name,
          });
        });

        this.setState({ authors: authorsOptions });
        return authorsOptions;
      })
      .catch((err) => {
        return this.state.authors;
      });
  };

  handleAuthorChange = (arr) => {
    let filters = { ...this.state.filters };

    filters.author = arr ? arr : [];
    this.setState({ filters });
  };

  handleInputChange = (e) => {
    let { name, value } = e.target;
    let filters = { ...this.state.filters };

    filters[name] = value;
    this.setState({ filters });
  };

  handleRouteChange = (arr) => {
    let filters = { ...this.state.filters };

    filters.routes = arr ? arr : [];
    this.setState({ filters });
  };

  clear = () => {
    this.setState({ dateFilterType: "range", filters: {} });
    this.props.setFilters({});
  };

  save = () => {
    let filters = { ...this.state.filters };
    let newRoutes = [];

    if (filters.routes && filters.routes.length) {
      filters.routes.map((item) => {
        newRoutes.push(item.value);
      });

      filters.routes = newRoutes;
    }

    this.props.setFilters(filters);
  };

  shouldAllowReportGeneration = () => {
    let filters = { ...this.state.filters };

    if (filters.published_after && filters.published_before) {
      return true;
    }
    return false;
  };

  handleDateFilterTypeChange = (e) => {
    let type = e.target.value;
    let published_after = "";
    let published_before = "";

    // 2020-02-04

    if (type === "thisWeek") {
      published_after = moment().startOf("isoWeek").format("YYYY-MM-DD");
      published_before = moment().format("YYYY-MM-DD");
    } else if (type === "thisMonth") {
      published_after = moment().startOf("month").format("YYYY-MM-DD");
      published_before = moment().format("YYYY-MM-DD");
    } else if (type === "lastWeek") {
      published_after = moment()
        .subtract(1, "weeks")
        .startOf("isoWeek")
        .format("YYYY-MM-DD");
      published_before = moment()
        .subtract(1, "weeks")
        .endOf("isoWeek")
        .format("YYYY-MM-DD");
    } else if (type === "lastMonth") {
      published_after = moment()
        .subtract(1, "months")
        .startOf("month")
        .format("YYYY-MM-DD");
      published_before = moment()
        .subtract(1, "months")
        .endOf("month")
        .format("YYYY-MM-DD");
    }

    let filters = { ...this.state.filters };
    filters.published_after = published_after;
    filters.published_before = published_before;

    this.setState({ filters, dateFilterType: type });
  };

  render() {
    let routesOptions = [];

    this.props.routes.map((route) => {
      routesOptions.push({
        value: route.id,
        label: route.name,
      });
    });

    return (
      <div className="sd-filters-panel sd-filters-panel--border-right">
        <div className="side-panel side-panel--transparent side-panel--shadow-right">
          <div className="side-panel__header side-panel__header--border-b">
            <span className="side-panel__close">
              <IconButton
                icon="close-small"
                tooltip={{ text: "Close", flow: "left" }}
                onClick={this.props.toggle}
              />
            </span>
            <h3 className="side-panel__heading side-panel__heading--big">
              {gettext("Filter")}
            </h3>
          </div>
          <div className="side-panel__content">
            <div className="side-panel__content-block">
              <div className="form__row">
                <div className="sd-line-input sd-line-input--no-margin">
                  <label className="sd-line-input__label">{gettext("Category")}</label>
                  <MultiSelect
                    onSelect={(values) => this.handleRouteChange(values)}
                    options={routesOptions}
                    selectedOptions={
                      this.state.filters.routes ? this.state.filters.routes : []
                    }
                  />
                </div>
              </div>
              <div className="form__row">
                <div className="sd-line-input sd-line-input--no-margin">
                  <label className="sd-line-input__label">{gettext("Authors")}</label>
                  <AsyncMultiSelect
                    onSelect={(values) => this.handleAuthorChange(values)}
                    loadOptions={(inputValue) => this.loadAuthors(inputValue)}
                    selectedOptions={
                      this.state.filters.author ? this.state.filters.author : []
                    }
                  />
                </div>
              </div>
              <div className="form__row">
                <div className="sd-line-input sd-line-input--no-margin sd-line-input--is-select">
                  <label className="sd-line-input__label">{gettext("Date filter")}</label>
                  <select
                    className="sd-line-input__select"
                    onChange={this.handleDateFilterTypeChange}
                    name="route"
                    value={this.state.dateFilterType}
                  >
                    <option value="range">{gettext("Custom")}</option>
                    <option value="lastWeek">{gettext("Last week")}</option>
                    <option value="lastMonth">{gettext("Last month")}</option>
                    <option value="thisWeek">{gettext("This week")}</option>
                    <option value="thisMonth">{gettext("This month")}</option>
                  </select>
                </div>
              </div>

              {this.state.dateFilterType === "range" && (
                <div className="form__row form__row--flex">
                  <div className="form__row-item">
                    <div className="sd-line-input sd-line-input--no-margin">
                      <label className="sd-line-input__label">{gettext("From")}</label>
                      <DatePicker
                        value={
                          this.state.filters.published_after
                            ? moment(
                                this.state.filters.published_after,
                                "YYYY-MM-DD"
                              ).toDate()
                            : null
                        }
                        dateFormat="YYYY-MM-DD"
                        onChange={(date) => {
                          let stringDate = moment(date).format("YYYY-MM-DD");

                          this.handleInputChange({
                            target: {
                              name: "published_after",
                              value: stringDate,
                            },
                          });
                        }}
                      />
                    </div>
                  </div>
                  <div className="form__row-item">
                    <div className="sd-line-input sd-line-input--no-margin">
                      <label className="sd-line-input__label">{gettext("To")}</label>
                      <DatePicker
                        value={
                          this.state.filters.published_before
                            ? moment(
                                this.state.filters.published_before,
                                "YYYY-MM-DD"
                              ).toDate()
                            : null
                        }
                        dateFormat="YYYY-MM-DD"
                        onChange={(date) => {
                          let stringDate = moment(date).format("YYYY-MM-DD");
                          this.handleInputChange({
                            target: {
                              name: "published_before",
                              value: stringDate,
                            },
                          });
                        }}
                      />
                    </div>
                  </div>
                </div>
              )}
            </div>
          </div>
          <div className="side-panel__footer side-panel__footer--button-box">
            <div className="flex-grid flex-grid--boxed-small flex-grid--small-2">
              <Button text={gettext("Clear")} style="hollow" onClick={this.clear} />
              <Button text={gettext("Run Report")} type="primary" onClick={this.save} />
            </div>
            <div
              className="flex-grid flex-grid--boxed-small flex-grid--small-1"
              data-sd-tooltip={
                this.shouldAllowReportGeneration()
                  ? null
                  : gettext("Select date range please")
              }
              style={{ overflow: "visible" }}
            >
              <Button
                text={gettext("Generate Report")}
                type="primary"
                expanded={true}
                onClick={() =>
                  this.shouldAllowReportGeneration()
                    ? this.props.generateReport(this.state.filters)
                    : null
                }
                disabled={!this.shouldAllowReportGeneration()}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

FiltersPanel.propTypes = {
  toggle: PropTypes.func.isRequired,
  filters: PropTypes.object.isRequired,
  setFilters: PropTypes.func.isRequired,
  routes: PropTypes.array,
  publisher: PropTypes.object.isRequired,
  generateReport: PropTypes.func.isRequired,
};

export default FiltersPanel;
