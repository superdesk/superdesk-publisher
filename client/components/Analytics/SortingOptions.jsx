import React from "react";
import { IconButton } from "superdesk-ui-framework/react";
import Dropdown from "../UI/Dropdown";
import {gettext} from '../../superdeskApi';

const SortingOptions = ({ filters, setFilters }) => {
  const setSort = (value) => {
    filters.sort = value;
    setFilters(filters);
  };

  const setOrder = (value) => {
    filters.order = value;
    setFilters(filters);
  };

  return (
    <React.Fragment>
      <Dropdown
        button={
          <button className="dropdown__toggle">
            {filters.sort === "published_at" ? "Publish date" : "Page views"}
            <span className="dropdown__caret"></span>
          </button>
        }
      >
        <li>
          <button onClick={() => setSort("published_at")}>{gettext("Publish date")}</button>
        </li>
        <li>
          <button
            onClick={() => setSort("article_statistics.page_views_number")}
          >
            {gettext("Page views")}
          </button>
        </li>
      </Dropdown>
      {filters.order === "desc" ? (
        <IconButton
          icon="descending"
          tooltip={{ text: gettext("Descending"), flow: "bottom" }}
          onClick={() => setOrder("asc")}
        />
      ) : (
        <IconButton
          icon="ascending"
          tooltip={{ text: gettext("Ascending"), flow: "bottom" }}
          onClick={() => setOrder("desc")}
        />
      )}
    </React.Fragment>
  );
};

export default SortingOptions;
