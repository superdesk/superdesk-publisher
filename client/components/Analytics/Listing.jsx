import React from "react";
import PropTypes from "prop-types";
import {gettext} from '../../superdeskApi';

import VirtualizedList from "../generic/VirtualizedList";
import ArticleItem from "./ArticleItem";

const Listing = props => {
  return (
    <div className="sd-column-box__main-column-inner sd-d-flex">
      <div
        className="sd-flex-table sd-table--shadowed sd-table--action-hover"
        style={{ flexGrow: "1", position: "relative" }}
      >
        <div className="sd-flex-table__row sd-flex-table--head">
          <div className="sd-flex-table__cell sd-flex-grow">
            <div>{gettext("Title")}</div>
          </div>
          <div className="sd-flex-table__cell">&nbsp;</div>
          <div className="sd-flex-table__cell">
            <div>{gettext("Page Views")}</div>
          </div>
          {/* <div className="sd-flex-table__cell">
          <div>Click Rate</div>
        </div>
        <div className="sd-flex-table__cell">
          <div>Impressions</div>
        </div> */}
        </div>
        {!props.articles.loading &&
          props.articles.items &&
          !props.articles.items.length && (
            <p style={{ padding: "1em", textAlign: "center" }}>
              {gettext("No results found")}
            </p>
          )}
        {!props.loading && (
          <VirtualizedList
            hasNextPage={
              props.articles.totalPages > props.articles.page ? true : false
            }
            isNextPageLoading={props.articles.loading}
            loadNextPage={props.queryArticles}
            items={props.articles.items}
            itemSize={props.articles.itemSize}
            ItemRenderer={ArticleItem}
            heightSubtract={46}
          />
        )}
      </div>
    </div>
  );
};

Listing.propTypes = {
  articles: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  queryArticles: PropTypes.func.isRequired
};

export default Listing;
