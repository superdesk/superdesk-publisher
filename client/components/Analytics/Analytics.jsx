import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import SitesSideNav from "../generic/SitesSideNav";
import FiltersPanel from "./FiltersPanel";
import VirtualizedList from "../generic/VirtualizedList";
import ArticleItem from "./ArticleItem";

class Analytics extends React.Component {
  constructor(props) {
    super(props);

    this._isMounted = false;

    this.state = {
      loading: true,
      tenantsNavOpen: false,
      filters: {},
      filtersOpen: false,
      sites: [],
      routes: [],
      selectedSite: null,
      articles: {}
    };
  }

  componentDidMount() {
    this._isMounted = true;

    this.props.publisher
      .setToken()
      .then(this.props.publisher.querySites)
      .then(sites => {
        let tenantsNavOpen = sites.length > 1 ? true : false;
        this.setState({ sites, tenantsNavOpen });

        if (this.props.tenant) {
          let site = sites.find(site => site.code === this.props.tenant);
          if (site) this.setTenant(site);
        } else if (sites[0]) {
          this.setTenant(sites[0]);
        }
      });
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  setTenant = site => {
    this.props.publisher.setTenant(site);
    let articles = {
      items: [],
      page: 0,
      totalPages: 1,
      loading: false,
      itemSize: 50
    };
    this.setState({
      selectedSite: site,
      articles,
      loading: false,
      filters: {}
    });
    this._queryRoutes();
  };

  _queryRoutes = () => {
    this.props.publisher.queryRoutes({ type: "collection" }).then(routes => {
      if (this._isMounted) this.setState({ routes });
    });
  };

  _queryArticles = () => {
    let articles = this.state.articles;
    articles.loading = true;
    this.setState({ articles }, () => {
      let params = {};
      params.limit = 20;
      params["sorting[published_at]"] = "desc";
      params.status = "published";
      params.page = this.state.articles.page + 1;

      if (this.state.filters.author && this.state.filters.author.length) {
        params["author[]"] = this.state.filters.author;
      }

      if (this.state.filters.route && this.state.filters.route.length) {
        params["route[]"] = this.state.filters.route;
      }

      if (
        this.state.filters.published_before &&
        this.state.filters.published_before.length
      ) {
        params.published_before = this.state.filters.published_before;
      }

      if (
        this.state.filters.published_after &&
        this.state.filters.published_after.length
      ) {
        params.published_after = this.state.filters.published_after;
      }

      this.props.publisher.queryTenantArticles(params).then(response => {
        let articles = {
          page: response.page,
          totalPages: response.pages,
          items: [...this.state.articles.items, ...response._embedded._items],
          loading: false,
          itemSize: this.state.articles.itemSize
        };
        if (this._isMounted) this.setState({ articles });
      });
    });
  };

  toggleTenantsNav = () => {
    let tenantsNavOpen = !this.state.tenantsNavOpen;
    this.setState({ tenantsNavOpen });
  };

  toggleFilters = () => {
    let filtersOpen = !this.state.filtersOpen;
    let articles = { ...this.state.articles };
    articles.itemSize = filtersOpen ? 100 : 50;
    this.setState({ filtersOpen, articles });
  };

  setFilters = filters => {
    let articles = {
      items: [],
      page: 0,
      totalPages: 1,
      loading: false,
      itemSize: 100
    };

    this.setState({ loading: true }, () =>
      this.setState({ filters, articles, loading: false })
    );
  };

  render() {
    return (
      <React.Fragment>
        {this.state.loading && <div className="sd-loader" />}

        <div
          className={classNames("sd-page-content__content-block", {
            "open-filters": this.state.filtersOpen
          })}
        >
          <div className="subnav">
            <a
              href="#/publisher"
              className="navbtn navbtn--left"
              sd-tooltip="Dashboard"
              flow="right"
            >
              <i className="icon-arrow-left" />
            </a>
            <h3 className="subnav__page-title">
              Content Analytics
              {this.state.selectedSite && this.state.selectedSite.name && (
                <span> / {this.state.selectedSite.name}</span>
              )}
            </h3>
          </div>
          <div
            className={classNames("sd-column-box--3", {
              "content-nav-closed": !this.state.tenantsNavOpen
            })}
          >
            <SitesSideNav
              sites={this.state.sites}
              selectedSite={this.state.selectedSite}
              setTenant={site => this.setTenant(site)}
              toggle={this.toggleTenantsNav}
              open={this.state.tenantsNavOpen}
            />
            <div className="flex-grid flex-grid--grow flex-grid--small-1">
              <div className="flex-grid__item flex-grid__item--d-flex flex-grid__item--column">
                <div className="subnav subnav--lower-z-index">
                  <button
                    className={classNames(
                      "navbtn navbtn--left navbtn--darker",
                      { "navbtn--active": this.state.filtersOpen }
                    )}
                    onClick={this.toggleFilters}
                  >
                    <i className="icon-filter-large" />
                  </button>
                </div>

                <div className="sd-column-box--3">
                  <FiltersPanel
                    toggle={this.toggleFilters}
                    filters={this.state.filters}
                    setFilters={filters => this.setFilters(filters)}
                    routes={this.state.routes}
                  />
                  <div className="sd-column-box__main-column-inner sd-d-flex">
                    <div
                      className="sd-flex-table sd-table--shadowed sd-table--action-hover"
                      style={{ flexGrow: "1", position: "relative" }}
                    >
                      <div className="sd-flex-table__row sd-flex-table--head">
                        <div className="sd-flex-table__cell sd-flex-grow">
                          <div>Title</div>
                        </div>
                        <div className="sd-flex-table__cell">&nbsp;</div>
                        <div className="sd-flex-table__cell">
                          <div>Page Views</div>
                        </div>
                        {/* <div className="sd-flex-table__cell">
                          <div>Click Rate</div>
                        </div>
                        <div className="sd-flex-table__cell">
                          <div>Impressions</div>
                        </div> */}
                      </div>
                      {!this.state.articles.loading &&
                        this.state.articles.items &&
                        !this.state.articles.items.length && (
                          <p style={{ padding: "1em", textAlign: "center" }}>
                            No results found
                          </p>
                        )}
                      {!this.state.loading && (
                        <VirtualizedList
                          hasNextPage={
                            this.state.articles.totalPages >
                            this.state.articles.page
                              ? true
                              : false
                          }
                          isNextPageLoading={this.state.articles.loading}
                          loadNextPage={this._queryArticles}
                          items={this.state.articles.items}
                          itemSize={this.state.articles.itemSize}
                          ItemRenderer={ArticleItem}
                          heightSubtract={46}
                        />
                      )}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

Analytics.propTypes = {
  tenant: PropTypes.string,
  publisher: PropTypes.object.isRequired
};

export default Analytics;
