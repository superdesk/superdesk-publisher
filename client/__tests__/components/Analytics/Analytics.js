import React from "react";
import Analytics from "../../../components/Analytics/Analytics";
import { render, waitForElement } from "@testing-library/react";

jest.mock("react-virtualized/styles.css", () => {
  return {};
});

import Publisher from "../../../__mocks__/publisher";

const publisher = new Publisher();

describe("Analytics/Analytics", () => {
  it("renders correctly", async () => {
    const { container, getByText } = render(
      <Analytics publisher={publisher} />
    );
    let title = await waitForElement(() => getByText("Content Analytics"));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("loads tenants", async () => {
    const { getByText } = render(<Analytics publisher={publisher} />);
    await waitForElement(() => getByText("Site1"));
  });
});
