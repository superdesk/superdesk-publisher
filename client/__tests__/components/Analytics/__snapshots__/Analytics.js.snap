// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics/Analytics renders correctly 1`] = `
<div
  class="sd-page-content__content-block"
>
  <div
    class="subnav"
  >
    <a
      class="navbtn navbtn--left"
      flow="right"
      href="#/publisher"
      sd-tooltip="Dashboard"
    >
      <i
        class="icon-arrow-left"
      />
    </a>
    <h3
      class="subnav__page-title"
    >
      Content Analytics
      <span>
         / 
        Site1
      </span>
    </h3>
  </div>
  <div
    class="sd-column-box--3"
  >
    <div
      class="nav-tabs-vertical nav-tabs-vertical--closed"
    >
      <ul
        class="nav-tabs-vertical__list"
      >
        <li
          class="nav-tabs-vertical__tab"
        >
          <button
            class="nav-tabs-vertical__link"
          >
            Website
          </button>
        </li>
      </ul>
    </div>
    <div
      class="sd-content-navigation-panel sd-content-navigation-panel--border-right"
    >
      <div
        class="subnav subnav--padded subnav--lower-z-index"
      >
        <h3
          class="sd-content-nav-title sd-content-nav-title--uppercase"
        >
          Website
        </h3>
        <a
          class="icn-btn ml-auto"
        >
          <i
            class="icon-close-small"
          />
        </a>
      </div>
      <nav
        class="sd-content-nav"
      >
        <ul>
          <li>
            <a
              class="sd-content-nav__btn sd-content-nav__btn--active"
              style="cursor: pointer;"
            >
              Site1
            </a>
          </li>
          <li>
            <a
              class="sd-content-nav__btn"
              style="cursor: pointer;"
            >
              Site2
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="flex-grid flex-grid--grow flex-grid--small-1"
    >
      <div
        class="flex-grid__item flex-grid__item--d-flex flex-grid__item--column"
      >
        <div
          class="subnav subnav--lower-z-index"
        >
          <button
            class="navbtn navbtn--left navbtn--darker"
          >
            <i
              class="icon-filter-large"
            />
          </button>
        </div>
        <div
          class="sd-column-box--3"
        >
          <div
            class="sd-filters-panel sd-filters-panel--border-right"
          >
            <div
              class="side-panel side-panel--transparent side-panel--shadow-right"
            >
              <div
                class="side-panel__header side-panel__header--border-b"
              >
                <a
                  class="icn-btn side-panel__close"
                  flow="left"
                  sd-tooltip="Close filters"
                >
                  <i
                    class="icon-close-small"
                  />
                </a>
                <h3
                  class="side-panel__heading side-panel__heading--big"
                >
                  Filter
                </h3>
              </div>
              <div
                class="side-panel__content"
              >
                <div
                  class="side-panel__content-block"
                >
                  <div
                    class="form__row"
                  >
                    <div
                      class="sd-line-input sd-line-input--no-margin sd-line-input--is-select"
                    >
                      <label
                        class="sd-line-input__label"
                      >
                        Category
                      </label>
                      <select
                        class="sd-line-input__select"
                        name="route"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="1"
                        >
                          route1
                        </option>
                        <option
                          value="2"
                        >
                          route2
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="form__row"
                  >
                    <div
                      class="sd-line-input sd-line-input--no-margin"
                    >
                      <label
                        class="sd-line-input__label"
                      >
                        Author
                      </label>
                      <input
                        class="sd-line-input__input"
                        name="author"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form__row form__row--flex"
                  >
                    <div
                      class="sd-line-input sd-line-input--no-margin"
                    >
                      <label
                        class="sd-line-input__label"
                      >
                        Published after
                      </label>
                      <input
                        class="sd-line-input__input"
                        name="published_after"
                        type="date"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form__row form__row--flex"
                  >
                    <div
                      class="sd-line-input sd-line-input--no-margin"
                    >
                      <label
                        class="sd-line-input__label"
                      >
                        Published before
                      </label>
                      <input
                        class="sd-line-input__input"
                        name="published_before"
                        type="date"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="side-panel__footer side-panel__footer--button-box"
              >
                <div
                  class="flex-grid flex-grid--boxed-small flex-grid--small-2"
                >
                  <a
                    class="btn btn--hollow"
                    data-testid="filterClear"
                  >
                    Clear
                  </a>
                  <a
                    class="btn btn--primary"
                    data-testid="filterSave"
                  >
                    Filter
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sd-column-box__main-column-inner sd-d-flex"
          >
            <div
              class="sd-flex-table sd-table--shadowed sd-table--action-hover"
              style="flex-grow: 1; position: relative;"
            >
              <div
                class="sd-flex-table__row sd-flex-table--head"
              >
                <div
                  class="sd-flex-table__cell sd-flex-grow"
                >
                  <div>
                    Title
                  </div>
                </div>
                <div
                  class="sd-flex-table__cell"
                >
                  Â 
                </div>
                <div
                  class="sd-flex-table__cell"
                >
                  <div>
                    Page Views
                  </div>
                </div>
              </div>
              <p
                style="padding: 1em; text-align: center;"
              >
                No results found
              </p>
              <div
                style="overflow: visible; height: 0px; width: 0px;"
              />
              <div
                class="resize-triggers"
              >
                <div
                  class="expand-trigger"
                >
                  <div
                    style="width: 1px; height: 1px;"
                  />
                </div>
                <div
                  class="contract-trigger"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
